### PR TITLE
Cumulus 1732

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-1732: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1732)

This PR is a re-open of a PR that was in review prior to the PR purge, and addresses the concerns raised therein by @markdboyd.   Reviewers should get a copy of the original review content for context.

### CODE REVIEW

File: tasks/python-reference-activity/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# @cumulus/python-reference-activity
+
+This 'task' is a reference activity implementation that is included with Cumulus to allow integration testing of the `Cumulus message adapter` and [`cumulus-process-py`](https://github.com/nasa/cumulus-process-py) with a deployed activity.
+
+## Development
+
+Developmental use of this lambda is intended to be simple - the processing activity runs an activity through the CMA and returns a static processing output, integration tests can be then built against the `Reference Activity` step in the `PythonProcess` workflow.
```
Comment (markdboyd) [inline tasks/python-reference-activity/README.md]: Nit: I would link `PythonProcess` here as well. I would also use this link style

```
[I'm a reference-style link][Arbitrary case-insensitive reference text]

... 

[arbitrary case-insensitive reference text]: https://www.mozilla.org
```

That way you can define the target for links that repeat throughout the document

File: tasks/python-reference-activity/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# @cumulus/python-reference-activity
+
+This 'task' is a reference activity implementation that is included with Cumulus to allow integration testing of the `Cumulus message adapter` and [`cumulus-process-py`](https://github.com/nasa/cumulus-process-py) with a deployed activity.
```
Comment (markdboyd) [inline tasks/python-reference-activity/README.md]:
```suggestion
This 'task' is a reference activity implementation that is included with Cumulus to allow integration testing of the `cumulus-message-adapter` and [`cumulus-process-py`](https://github.com/nasa/cumulus-process-py) with a deployed activity.
```

Comment (markdboyd) [reply tasks/python-reference-activity/README.md]: `cumulus-message-adapter` should be linked

File: tasks/python-reference-activity/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+### Build
+
+To update the container for a PR, you should run:
+
+```bash
+docker build -t cumuluss/cumulus-process-activity:{VERSION} .
+docker push cumuluss/cumulus-process-activity:{VERSION}
+```
+
+Then update [`python_reference_workflow.tf`](https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/python_reference_workflow.tf) `python_processing_service` resource to utilize the correct image reference.
```
Comment (markdboyd) [inline tasks/python-reference-activity/README.md]:
```suggestion
Then update the  `python_processing_service` resource in [`python_reference_workflow.tf`](https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/python_reference_workflow.tf) to utilize the correct image reference.
```

File: example/spec/parallel/pythonReferenceTests/PythonReferenceSpec.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  it('contains the expected output from the PythonReferenceTask', async () => {
+    const stepOutput = await lambdaStep.getStepOutput(workflowExecution.executionArn, 'PythonReferenceTask');
+    const expectedPayload = {
+      inputData: initialPayload.initialData,
+      configInputData: { key1: 'injectedData' },
+      newData: { newKey1: 'newData1' }
+    };
+
+    expect(stepOutput.payload).toEqual(expectedPayload);
+    expect(Object.keys(stepOutput)).toContain('meta');
```
Comment (markdboyd) [inline example/spec/parallel/pythonReferenceTests/PythonReferenceSpec.js]: So these assertions just prove that the CMA is working properly and not overwriting unexpected keys?

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## [Unreleased]
 
+### Added
+
+- **CUMULUS-1732**
+  - Added Python task/activity workflow and integration test (`PythonReferenceSpec`) to test `cumulus-message-adapter-py`and `cumulus-process-py` integration.
```
Comment (markdboyd) [inline CHANGELOG.md]:
```suggestion
  - Added Python task/activity workflow and integration test (`PythonReferenceSpec`) to test `cumulus-message-adapter-python`and `cumulus-process-py` integration.
```

File: example/cumulus-tf/ecs_hello_world_workflow.tf [hunk 6]
```diff
@@ -12,7 +12,7 @@ module "hello_world_service" {
 
   cluster_arn                           = module.cumulus.ecs_cluster_arn
   desired_count                         = 1
-  image                                 = "cumuluss/cumulus-ecs-task:1.4.0"
+  image                                 = "jlkovarik/cumulus-ecs-task:1.5.1"
```
Comment (markdboyd) [inline example/cumulus-tf/ecs_hello_world_workflow.tf]: Are we not intending to publish this to `cumuluss`?

File: tasks/python-reference-task/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# @cumulus/python-reference-task
+
+This 'task' is a reference task that is included with Cumulus to allow integration testing of the [`Cumulus message adapter`](https://github.com/nasa/cumulus-message-adapter) with a built python lambda.
+
+## Use
+
+Developmental use of this lambda is intended to be simple - the task returns a static processing output, integration tests can be then built against the `Reference Task` step in the (`PythonReference`)[https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/python_reference_workflow.tf] workflow.
```
Comment (markdboyd) [inline tasks/python-reference-task/README.md]: This is referred to as the `PythonProcess` workflow in the other README. I'm fine with either, just want to be consistent

File: tasks/python-reference-task/README.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+# @cumulus/python-reference-task
+
+This 'task' is a reference task that is included with Cumulus to allow integration testing of the [`Cumulus message adapter`](https://github.com/nasa/cumulus-message-adapter) with a built python lambda.
```
Comment (markdboyd) [inline tasks/python-reference-task/README.md]:
```suggestion
This 'task' is a reference task that is included with Cumulus to allow integration testing of the [`cumulus-message-adapter`](https://github.com/nasa/cumulus-message-adapter) with a built python lambda.
```

File: tasks/python-reference-task/README.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+## Use
+
+Developmental use of this lambda is intended to be simple - the task returns a static processing output, integration tests can be then built against the `Reference Task` step in the (`PythonReference`)[https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/python_reference_workflow.tf] workflow.
+
+## Development
+
+Updates should generally consist of updates to the included `requirements.txt`, as the purpose of this task is to ensure compatibility with updates to the [cumulus-message-adapter-python](https://github.com/nasa/cumulus-message-adapter-python) client library and the [cumulus-message-adapter](https://github.com/nasa/cumulus-message-adapter) deployed with Cumulus via the CMA lambda layer ([Cumulus-message-adatper-python](https://github.com/nasa/cumulus-message-adapter-python) will utilize the layer added to the lambda by default if `CMA_DIR` is set).
```
Comment (markdboyd) [inline tasks/python-reference-task/README.md]:
```suggestion
Updates should generally consist of updates to the included `requirements.txt`, as the purpose of this task is to ensure compatibility with updates to the [cumulus-message-adapter-python](https://github.com/nasa/cumulus-message-adapter-python) client library and the [cumulus-message-adapter](https://github.com/nasa/cumulus-message-adapter) deployed with Cumulus via the CMA lambda layer ([cumulus-message-adatper-python](https://github.com/nasa/cumulus-message-adapter-python) will utilize the layer added to the lambda by default if `CMA_DIR` is set).
```

Again, I recommend this link style for these READMEs so you only have to set link targets once: `[I'm a reference-style link][Arbitrary case-insensitive reference text]`

File: example/cumulus-tf/ecs_hello_world_workflow.tf (hunk 6 above)
Comment (Jkovarik) [reply example/cumulus-tf/ecs_hello_world_workflow.tf]: Yes, just as soon as that PR is complete.

Comment (Jkovarik) [reply example/cumulus-tf/ecs_hello_world_workflow.tf]: 03e55b6d7

File: example/spec/parallel/pythonReferenceTests/PythonReferenceSpec.js (hunk 4 above)
Comment (Jkovarik) [reply example/spec/parallel/pythonReferenceTests/PythonReferenceSpec.js]: Yes, exactly.    We could do a more thorough evaluation of *what* is in meta/cumulus_meta, but that's not strictly constrained to this test.

File: tasks/python-reference-activity/README.md (hunk 2 above)
Comment (Jkovarik) [reply tasks/python-reference-activity/README.md]: 5d7678a8f

File: tasks/python-reference-activity/README.md (hunk 1 above)
Comment (Jkovarik) [reply tasks/python-reference-activity/README.md]: f416c8ed8

File: tasks/python-reference-activity/README.md (hunk 3 above)
Comment (Jkovarik) [reply tasks/python-reference-activity/README.md]: Went with f416c8ed8 for consistency, then 77857f5ed

File: tasks/python-reference-task/README.md (hunk 8 above)
Comment (Jkovarik) [reply tasks/python-reference-task/README.md]: Will update, but worth noting in the original review you suggested [Cumulus message adapter].  I like this better for the links, but without the tics - reason being the tics obscure the link.

Comment (Jkovarik) [reply tasks/python-reference-task/README.md]: Updated, made links/tics more consistent fc63e472d

File: tasks/python-reference-task/README.md (hunk 7 above)
Comment (Jkovarik) [reply tasks/python-reference-task/README.md]: 👍  22b5a0612

File: tasks/python-reference-activity/README.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# @cumulus/python-reference-activity
+
+[`python_reference_workflow`]: https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/python_reference_workflow.tf
+[`cumulus-process-py`]: https://github.com/nasa/cumulus-process-py
+
+This 'task' is a reference activity implementation that is included with Cumulus to allow integration testing of the [`cumulus-message-adapter`](`https://github.com/nasa/cumulus-message-adapter) and [`cumulus-process-py`] with a deployed activity.
```
Comment (markdboyd) [inline tasks/python-reference-activity/README.md]:
```suggestion
This 'task' is a reference activity implementation that is included with Cumulus to allow integration testing of the [`cumulus-message-adapter`](https://github.com/nasa/cumulus-message-adapter) and [`cumulus-process-py`] with a deployed activity.
```

First link wasn't working without this change

File: tasks/python-reference-activity/README.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+## Development
+
+Developmental use of this lambda is intended to be simple - the processing activity runs an activity through the CMA and returns a static processing output, integration tests can be then built against the `Reference Activity` step in the [`python_reference_workflow`].
+### Requirements
```
Comment (markdboyd) [inline tasks/python-reference-activity/README.md]:
```suggestion

### Requirements
```

File: tasks/python-reference-task/README.md (hunk 8 above)
Comment (markdboyd) [reply tasks/python-reference-task/README.md]: My bad. Yeah, I like this better

**Pull request merged**